lsp-methods.el

```diff
@@ -940,7 +940,15 @@ https://github.com/Microsoft/language-server-protocol/blob/master/protocol.md#co
       :annotation-function #'lsp--annotate
       :display-sort-function #'lsp--sort-completions)))
 
-;;; TODO: implement completionItem/resolve
+(defun lsp--resolve-completion (comp)
+  (lsp--cur-workspace-check)
+  (lsp--send-changes lsp--cur-workspace)
+  (unless (gethash "resolveProvider" (lsp--capability "completionProvider"))
+    (error "ResolveProvider not found"))
```

Comment:
It seems unnecessary to error out when the server does not provide `resolve` support. Returning `nil` would suffice.

Reply:
This is already changed in 27ba5409dc86949d9d0ce751ab15f26a6c7f5ae0.

I don't think retuning nil is a good idea. This makes callers worry about the capability. Always returning a completion item is a more consistent behavior.

Reply:
Indeed, returning the item is much better.